NGitLab/Impl/GroupVariableClient.cs

```diff
@@ -1,11 +1,12 @@
-﻿using NGitLab.Models;
+﻿using System;
+using NGitLab.Models;
 
 namespace NGitLab.Impl;
 
 internal sealed class GroupVariableClient : VariableClient, IGroupVariableClient
 {
-    public GroupVariableClient(API api, GroupId groupId)
-        : base(api, $"{Group.Url}/{groupId.ValueAsUriParameter()}")
+    public GroupVariableClient(API api, GroupId groupId, string environmentScope = null)
+        : base(api, string.IsNullOrWhiteSpace(environmentScope) ? $"{Group.Url}/{groupId.ValueAsUriParameter()}" : $"{Group.Url}/{groupId.ValueAsUriParameter()}?filter[environment_scope]={Uri.EscapeDataString(environmentScope!)}")
```

Comment:
I would avoid logic in the constructor.

Can the presence of the querystring be determined in the methods directly?

Reply:
You're totally right. I reverted the change ...